src/applications/oscilloscope.py

```diff
@@ -1,117 +1,247 @@
-import time
 import argparse
 import collections
 import tkinter as Tk
 import logging
+from typing import Any
 
 import numpy as np
-from matplotlib.lines import Line2D
 import matplotlib.pyplot as plt
 import matplotlib.animation as animation
 
-from matplotlib.figure import Figure
 from matplotlib.backends.backend_tkagg import FigureCanvasTkAgg, NavigationToolbar2Tk
 
-import nidaqmx
-
-import qt3utils.nidaq
 import qt3utils.datagenerators as datasources
+from qt3utils.math_utils import get_rolling_mean
```

Comment:
Usually, I don't prefer this type of import except for in particular cases where the library usage is ubiquitous throughout the industry (`import matplotlib.pyplot as plt` for example).  I think its always clearer to use 
the full import statement when you can. Although, I can see that I've clearly violated this preference above.**

When using a full import path, it's then clear where that function is defined. This can be a problem though if it creates long lines of code. So, it can be a toss up. However, in this case, 'get_rolling_mean' is defined locally in `ScopeFigure` and in an external module, so leaving the full import path makes it clearer to understand, I think. 



My preference is 
`import qt3utils.math_utils`

and then call

`qt3utils.math_utils.get_rolling_mean` below on line 220. 

------


** I think the line`import qt3utils.datagenerators as datasource` should be changed to `import qt3utils.datagenerators` and then use `qt3utils.datagenerators.RandomRateCounter()` and `qt3utils.datagenerators.NiDaqDigitalInputRateCounter` below

Reply:
Sounds good. I will try to follow this rule, but I think we can add a bit of slack on the rule for when a method is used too many times and the code line gets quite long